sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/datastore/DataStoreV1TableProvider.java

```diff
@@ -39,15 +43,14 @@
  * }</pre>
  */
 @AutoService(TableProvider.class)
-public class DataStoreV1TableProvider extends InMemoryMetaTableProvider {
-
+public class DataStoreV1TableProvider extends SchemaIOTableProviderWrapper {
   @Override
-  public String getTableType() {
-    return "datastoreV1";
+  public SchemaIOProvider getSchemaIOProvider() {
+    return new DataStoreV1SchemaIOProvider();
   }
 
   @Override
-  public BeamSqlTable buildBeamSqlTable(Table table) {
-    return new DataStoreV1Table(table);
+  public String getTableType() {
+    return "datastoreV1";
```

Comment:
Should this also have an implementation for getTableStatistics? It looks like DataStoreV1Table had a non-standard implementation:
```
  @Override	
  public BeamTableStatistics getTableStatistics(PipelineOptions options) {	
    long count =	
        DatastoreIO.v1().read().withProjectId(projectId).getNumEntities(options, kind, null);	

    if (count < 0) {	
      return BeamTableStatistics.BOUNDED_UNKNOWN;	
    }	

    return BeamTableStatistics.createBoundedTableStatistics((double) count);	
  }
```

Reply:
Done.